yarn-project/end-to-end/src/devnet/e2e_devnet.test.ts

```diff
@@ -0,0 +1,226 @@
+  let l1ChainId: number;
+  let feeJuiceL1: EthAddress;
+  let teardown: () => void | Promise<void>;
+
+  beforeAll(async () => {
+    logger = getLogger();
+
+    if (!ETHEREUM_HOST) {
+      throw new Error('ETHEREUM_HOST must be set');
+    }
+
+    if (!AZTEC_CLI) {
+      throw new Error('AZTEC_CLI must be set');
+    }
+
+    if (!FAUCET_URL) {
+      throw new Error('FAUCET_URL must be set');
+    }
+
+    logger.info(`Using AZTEC_CLI: ${AZTEC_CLI}`);
+
+    if (AZTEC_NODE_URL) {
+      logger.info(`Using AZTEC_NODE_URL: ${AZTEC_NODE_URL}`);
+      const node = createAztecNodeClient(AZTEC_NODE_URL);
+      const svc = await setupPXEService(node);
+      pxe = svc.pxe;
+
+      const port = await getPort();
+      pxeUrl = `http://localhost:${port}`;
+      // start a server for the CLI to talk to
+      const server = startHttpRpcServer('pxe', pxe, createPXERpcServer, port);
+
+      teardown = async () => {
+        const { promise, resolve, reject } = promiseWithResolvers<void>();
+        server.close(e => (e ? reject(e) : resolve()));
+        await promise;
+
+        await svc.teardown();
+      };
+    } else if (PXE_URL) {
+      logger.info(`Using PXE_URL: ${PXE_URL}`);
+      pxe = createPXEClient(PXE_URL);
+      pxeUrl = PXE_URL;
+      teardown = () => {};
+    } else {
+      throw new Error('AZTEC_NODE_URL or PXE_URL must be set');
+    }
+
+    ({
+      l1ChainId,
+      l1ContractAddresses: { gasTokenAddress: feeJuiceL1 },
+    } = await pxe.getNodeInfo());
+    logger.info(`PXE instance started`);
+  });
+
+  afterAll(async () => {
+    await teardown();
+  });
+
+  it('deploys an account while paying with FeeJuice', async () => {
+    const l1Account = await cli<{ privateKey: string; address: string }>('create-l1-account');
+    const l2Account = getSchnorrAccount(pxe, Fr.random(), Fq.random(), Fr.random());
+
+    await expect(getL1Balance(l1Account.address)).resolves.toEqual(0n);
+    await expect(getL1Balance(l1Account.address, feeJuiceL1)).resolves.toEqual(0n);
+
+    await cli('drip-faucet', { 'faucet-url': FAUCET_URL!, token: 'eth', address: l1Account.address });
+    await cli('drip-faucet', { 'faucet-url': FAUCET_URL!, token: 'fee_juice', address: l1Account.address });
+
+    await expect(getL1Balance(l1Account.address)).resolves.toBeGreaterThan(0n);
+    await expect(getL1Balance(l1Account.address, feeJuiceL1)).resolves.toBeGreaterThan(0n);
+
+    const amount = 1_000_000_000_000n;
+    const { claimAmount, claimSecret } = await cli<{ claimAmount: string; claimSecret: string }>(
+      'bridge-fee-juice',
+      [amount, l2Account.getAddress()],
+      {
+        'l1-rpc-url': ETHEREUM_HOST!,
+        'l1-chain-id': l1ChainId.toString(),
+        'l1-private-key': l1Account.privateKey,
+        'rpc-url': pxeUrl,
+      },
+    );
+
+    // TODO (alexg) remove this once sequencer builds blocks continuously
+    await advanceChainForBridgingToWork();
+
+    const txReceipt = await l2Account
```

Comment:
I should run this with the cli

Reply:
Update: the CLI process doesn't exit in this case. Not sure why that is so I've commented the code out.